src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs

```diff
@@ -77,11 +77,21 @@ private async Task ReloadAsync(CancellationToken cancellationToken)
             }
         }
 
-        private static bool IsJson(string str)
+        private static bool TryParseJson(string data, out JToken? jToken)
         {
+            jToken = null;
+
+            var trimmedData = data.Trim();
```

Comment:
Rather than trimming and using FirstOrDefault(), I'd check for emptiness and shortcut to false.

so, something like
```
if (string.IsNullOrEmpty(data) || (data[0] != '[' && data[0] != '{')) return false;
```

Reply:
> Rather than trimming and using FirstOrDefault(), I'd check for emptiness and shortcut to false.
> 
> so, something like
> 
> ```
> if (string.IsNullOrEmpty(data) || (data[0] != '[' && data[0] != '{')) return false;
> ```

I'd rather leave the trimming, just replaced Trim() with TrimStart().

Saving the string with the leading spaces is impossible with the AWS WEB console interface, but it's possible programmatically:

```
PutSecretValueResponse result = await _client.PutSecretValueAsync(new PutSecretValueRequest
{
	SecretId = secretEntry.ARN,
	SecretString = "     {testProp: 12345}"
});

GetSecretValueResponse secretValue = await _client
	.GetSecretValueAsync(new GetSecretValueRequest {SecretId = secretEntry.ARN}, cancellationToken)
	.ConfigureAwait(false);
	
Console.WriteLine("-----OUTPUT-----");	
Console.WriteLine(secretValue.SecretString);

//-----OUTPUT-----
//     {testProp: 12345}
```

@Kralizek, what do you think?

Reply:
Ok. Let's continue this way. Please add a test that proves that we handle leading spaces in the secret :)

Reply:
Done.